src/esp/gfx/GenericDrawable.cpp

```diff
@@ -54,14 +56,18 @@ void GenericDrawable::draw(const Mn::Matrix4& transformationMatrix,
     lightColors.emplace_back((*lightSetup_)[i].color);
   }
 
+  Mn::Color3 highLightColor = 0xa5c9ea_rgbf * 0.5;
   (*shader_)
-      .setAmbientColor(materialData_->ambientColor)
+      .setAmbientColor(selected_ ? highLightColor : materialData_->ambientColor)
```

Comment:
It feels a little weird for such application-specific selection logic to live in this GenericDrawable class, namely the choice of highlight color and the choice to override ambient color when highlighted. It feels more like the viewer class would like to make those decisions.

Maybe this is good enough for now and we can revisit it when we finalize our material API. Maybe the viewer will be able to easily swap in an alternate highlight material at that point.

Reply:
Feels quite weird for me as well. I'd say the main goal for most of the code is to be optimized for the training use case, and since object picking is not part of that use case, it's extra overhead that doesn't belong there.

In the magnum-player app I'm adding a temporary second drawable for the highlighted object, which I think is a better way. Again, the overhead of drawing the same object twice doesn't matter as much in this case because it's the viewer and we don't need it to run 1000 FPS.

Additionally, the extra drawable would then allows you to use a totally different shader and thus open the possibilities for wireframe visualization, normal visualization, topology/material inspection, overdraw / texture LOD visualization etc etc., like the latest magnum-player has.